### PR TITLE
VIH-9738 - Previous-Message-Cleared-When-New-Search-Starts

### PR DESCRIPTION
**Pull request by RosDiaf**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-9738

### Change description ###
The message 'Hearings have been updated.' needs to be cleared when:

- Allocated to user is selected
- Hearing is selected
- The same hearings is reallocated the message is cleared and auto refreshed when the other user is confirmed.

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ export class AllocateHearingsComponent implements OnInit {
     customIconTransform: Transform = { rotate: 45 };
     private filterSize = 20;
     dropDownUserLabelAllocateTo = 'Allocate to';
+    readonly HEARING_HAVE_BEEN_UPDATED = 'Hearings have been updated.';
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts:46]: we have a Constants file (in src/app/common/constants.ts) which we use for messages. Could we store it in there instead?

with an entry something like:

`AllocateHearings: {
        Confirmed: 'Hearings have been updated.'
    }`
    
but good idea nonetheless though, moving it into a constant :)

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts [hunk 2]
```diff
@@ -454,4 +454,47 @@ describe('AllocateHearingsComponent', () => {
             expect(clockIcon).toBeFalsy();
         });
     });
+
+    describe('Clear Hearing Updated Message', () => {
+        let HEARING_HAVE_BEEN_UPDATED: string;
+        let OTHER_MESSAGES: string;
+        beforeAll(() => {
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts:461]: `beforeAll()` shouldn't be needed here, think you can change the `let`s to

```
const HEARING_HAVE_BEEN_UPDATED: string = 'Hearings have been updated.';
const OTHER_MESSAGES: string = 'Other messages';
```

and then remove the `beforeAll`

**Review by oliver-scott — COMMENTED** [review]
looks good, added a couple of suggestions

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts (hunk 1 above)
Comment (RosDiaf) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts]: Yes, I was looking for such a file.
Confimation message moved to constants file now.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts (hunk 2 above)
Comment (RosDiaf) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts]: const added and beforeAll() removed

**Review by marcogagliardi — APPROVED** [review]
👌🏽

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**